### PR TITLE
end level logic

### PR DESCRIPTION
**Pull request by michaelDemertzi**

### CODE REVIEW

File: model/model.py [hunk 1]
```diff
@@ -57,8 +60,11 @@ def step(self, user_input: Optional[Direction]) -> None:
                 self._goto_next_state()
 
         elif self._curr_state == State.PEOPLE_OFF:
-            self._setup_next_state(State.PEOPLE_ON, STATE_TIME_PEOPLE_OFF_SECONDS)
             self._passengers.drop_off(self.get_current_floor())
+            if self._is_level_finished():
```
Comment (thbenschop) [inline model/model.py:64]: why checking in PEOPLE_OFF?

File: model/model.py [hunk 2]
```diff
@@ -85,6 +91,11 @@ def step(self, user_input: Optional[Direction]) -> None:
             self._setup_next_state(State.ARRIVING, STATE_TIME_MOVING_SECONDS)
             if self._state_timeout():
                 self._goto_next_state()
+
+        elif self._curr_state == State.LEVEL_COMPLETE:
+            # TODO exit level somehow
```
Comment (thbenschop) [inline model/model.py:96]: exit(-1)

File: view/console_view.py [hunk 3]
```diff
@@ -45,6 +51,13 @@ def print_game(self) -> None:
         game_text = '\n'.join(reversed(floors))
         self._writer.write(game_text)
 
+    def print_level_complete(self):
+        moves = self._bellhop_model.get_move_number()
+        if self._tips is None:
+            self._tips = round(random.random()*1e3, 2)
```
Comment (thbenschop) [inline view/console_view.py:57]: lol love the scoring mechanism

File: model/model.py (hunk 2 above)
Comment (thbenschop) [reply model/model.py:96]: Or hook up 'q' to exit

File: level_tests.py [hunk 4]
```diff
@@ -21,5 +21,6 @@ def test__get_next_event__moves_out_of_order__throws_exception(self, mock_level_
         mock_level_file_open.return_value.__next__ = lambda self: next(iter(self.readline, ''))
 
         filename = "filename.txt"
-        # mock_level_file_open.assert_called_once_with(filename, 'r') # I don't understand this but it's failing when I'm faily sure it shouldn't be
+        # mock_level_file_open.assert_called_once_with(filename, 'r')
```
Comment (screwed99) [inline level_tests.py:24]: Yo, uncomment this and put it after your assertRaises. You are asserting that the mock has been called before you've constructed the Level object (i.e. the file open hasn't been called yet)

File: view/console_view.py [hunk 5]
```diff
@@ -32,9 +34,13 @@ def __init__(self, bellhop_model: IBellhopViewer, model_vars: Dict[str, int], wr
         self._num_floors: int = model_vars['num_floors']
         self._capacity: int = model_vars['capacity']
         self._writer: PyGameViewTextWriter = writer
+        self._tips: Optional[int] = None
```
Comment (screwed99) [inline view/console_view.py:37]: Move 'tips' to model

**Review by screwed99 — CHANGES_REQUESTED** [review]

File: level_tests.py [hunk 6]
```diff
@@ -21,5 +21,6 @@ def test__get_next_event__moves_out_of_order__throws_exception(self, mock_level_
         mock_level_file_open.return_value.__next__ = lambda self: next(iter(self.readline, ''))
```
Comment (screwed99) [inline level_tests.py:21]: Move these level files to the level directory and update all the module imports

**Review by screwed99 — CHANGES_REQUESTED** [review]

File: model/model.py (hunk 1 above)
Comment (michaelDemertzi) [reply model/model.py:64]: just thought it made sense to do it after the consequences of your last move is processed. But agree it seems clunky

File: view/console_view.py (hunk 3 above)
Comment (michaelDemertzi) [reply view/console_view.py:57]: as fair as real life

**Pull request merged**